pkg/tsdb/influxdb/influxdb.go

```diff
@@ -23,12 +24,14 @@ import (
 var logger log.Logger = log.New("tsdb.influxdb")
 
 type Service struct {
+	Cfg      *setting.Cfg
 	im       instancemgmt.InstanceManager
 	features featuremgmt.FeatureToggles
 }
 
-func ProvideService(httpClient httpclient.Provider, features featuremgmt.FeatureToggles) *Service {
+func ProvideService(cfg *setting.Cfg, httpClient httpclient.Provider, features featuremgmt.FeatureToggles) *Service {
```

Comment:
You shouldn't use this as a dependency. Rather I think you should read the timeout value from HTTP options retrieved in newInstanceSettings?